src/runtime/composables/usePortal.ts

```diff
@@ -0,0 +1,24 @@
+import { inject, provide, computed, type Ref, type InjectionKey } from 'vue'
+
+export const portalTargetInjectionKey: InjectionKey<Ref<string | HTMLElement>> = Symbol('nuxt-ui.portal-target')
+
+export function usePortal(portal: Ref<string | HTMLElement | boolean | undefined>) {
+  const portalTarget = inject(portalTargetInjectionKey, undefined)
+
+  const to = computed(() => {
+    if (typeof portal.value === 'string' || portal.value instanceof HTMLElement) {
+      return portal.value
+    }
+
+    return portalTarget?.value ?? 'body'
+  })
+
+  const disabled = computed(() => typeof portal.value === 'boolean' ? !portal.value : false)
+
+  provide(portalTargetInjectionKey, computed(() => to.value))
+
+  return {
+    to: to.value,
+    disabled: disabled.value
+  }
```

Comment:
Maybe removing the .value access from the composable may prevent the reactivity from vanishing ? 

```suggestion
  return {
    to,
    disabled
  }
```

Reply:
You're right, I've put back the `computed` on top!